feat: Added support for using microdnf

vehagn opened the pull request:
Tested with [rockylinux:9-minimal](https://hub.docker.com/_/rockylinux) and [redhat/ubi9-minimal](https://hub.docker.com/r/redhat/ubi9-minimal/tags) container images which both only contain [`microdnf`](https://github.com/rpm-software-management/microdnf) instead of `dnf` (or `yum`).

riosje commented:
Hi @vehagn thank you so much for taking the time and propose those enhancements.
Do you mind to add `rockylinux9` and `ubi9` to the CI? so we can guarantee it will not break in the future.

vehagn commented:
@riosje I’ll add them tomorrow. The interesting images are the minimal ones that don’t have neither `dnf` nor `yum`.

vehagn commented:
Added a new CI-section for testing `microdnf`.

[Rocky Linux](https://rockylinux.org/) is "designed to be 100% bug-for-bug compatible with Red Hat Enterprise Linux".
[Red Hat Universal Base Image](https://catalog.redhat.com/software/base-images) (UBI) is "a collection of Open Container Initiative (OCI)-compliant, freely redistributable, container base operating system images".

I'm not 100% if it's necessary to test both Rocky Linux and UBI as they're supposed to be similar.

I'll also be happy to remove the line I added in the README.md file about Redhat 9-minimal if you think it shouldn't be there.

riosje reviewed (approved):
LGTM

riosje commented:
@vehagn  This is amazing, thank you so much for your contribution.
The scripts will be in sync in the next few hours.
